core/src/main/java/com/google/cloud/sql/core/LazyRefreshStrategy.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.sql.core;
+
+import com.google.common.util.concurrent.RateLimiter;
+import com.google.errorprone.annotations.concurrent.GuardedBy;
+import java.time.Instant;
+import java.util.function.Supplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** RefreshStrategy that implements the lazy refresh strategy. */
+public class LazyRefreshStrategy implements RefreshStrategy {
+  private final Logger logger = LoggerFactory.getLogger(LazyRefreshStrategy.class);
+
+  private final String name;
+  private final Supplier<ConnectionInfo> refreshOperation;
+
+  private final RateLimiter rateLimiter;
```

Comment:
Eno mentioned this in a below comment, but does the Lazy Refresh need a rate limiter? I would think no. Since the rate limiter is meant to slow background refresh calls I think we don't need it here.

We should not rate limit for lazy refresh as it may provide unintended behaviour.

Reply:
RateLimiter has been removed.